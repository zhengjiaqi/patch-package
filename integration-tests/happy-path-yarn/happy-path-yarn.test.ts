import { runIntegrationTest } from "../runIntegrationTest"
runIntegrationTest({
  projectName: "happy-path-yarn",
  shouldProduceSnapshots: true,
})
