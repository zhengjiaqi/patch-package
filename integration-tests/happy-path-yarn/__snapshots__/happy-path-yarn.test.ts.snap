// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test happy-path-yarn: making patch 1`] = `
"SNAPSHOT: making patch
patch-package 0.0.0
â€¢ Creating temporary folder
â€¢ Installing left-pad@1.1.3 with yarn
â€¢ Diffing your files with clean files
âœ” Created file patches/left-pad+1.1.3.patch

ðŸ’¡ left-pad is on GitHub! To draft an issue based on your patch run

    yarn patch-package left-pad --create-issue

END SNAPSHOT"
`;

exports[`Test happy-path-yarn: the patch looks like this 1`] = `
"SNAPSHOT: the patch looks like this
diff --git a/node_modules/left-pad/index.js b/node_modules/left-pad/index.js
index 26f73ff..b083802 100644
--- a/node_modules/left-pad/index.js
+++ b/node_modules/left-pad/index.js
@@ -22,9 +22,9 @@ var cache = [
 function leftPad (str, len, ch) {
   // convert \`str\` to \`string\`
   str = str + '';
-  // \`len\` is the \`pad\`'s length now
+  // \`len\` is the \`yarn\`'s length now
   len = len - str.length;
-  // doesn't need to pad
+  // doesn't need to yarn
   if (len <= 0) return str;
   // \`ch\` defaults to \`' '\`
   if (!ch && ch !== 0) ch = ' ';
@@ -32,12 +32,12 @@ function leftPad (str, len, ch) {
   ch = ch + '';
   // cache common use cases
   if (ch === ' ' && len < 10) return cache[len] + str;
-  // \`pad\` starts with an empty string
-  var pad = '';
+  // \`yarn\` starts with an empty string
+  var yarn = '';
   // loop
   while (true) {
-    // add \`ch\` to \`pad\` if \`len\` is odd
-    if (len & 1) pad += ch;
+    // add \`ch\` to \`yarn\` if \`len\` is odd
+    if (len & 1) yarn += ch;
     // divide \`len\` by 2, ditch the remainder
     len >>= 1;
     // \\"double\\" the \`ch\` so this operation count grows logarithmically on \`len\`
@@ -47,6 +47,6 @@ function leftPad (str, len, ch) {
     // \`len\` is 0, exit the loop
     else break;
   }
-  // pad \`str\`!
-  return pad + str;
+  // yarn \`str\`!
+  return yarn + str;
 }
END SNAPSHOT"
`;
